GradeCenter.API/Controllers/SchoolController.cs

```diff
@@ -9,34 +13,71 @@ namespace GradeCenter.API.Controllers
     public class SchoolController : ControllerBase
     {
         private readonly ISchoolService _schoolService;
+        private readonly IMapper _mapper;
 
-        public SchoolController(ISchoolService schoolService)
+        public SchoolController(ISchoolService schoolService, IMapper mapper)
         {
             _schoolService = schoolService;
+            _mapper = mapper;
+        }
+        /// <summary>
+        /// Reads all the existing School entities in the database.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("Read")]
+        public async Task<IActionResult> Read()
+        {
+            return Ok(_schoolService.Read());
         }
 
         /// <summary>
         /// Creates an object of type School and stores it in the database.
         /// </summary>
         /// <param name="requestModel"></param>
         /// <returns></returns>
+        [Authorize]
         [HttpPost("Create")]
         public async Task<IActionResult> Create(SchoolCreateRequestModel requestModel)
         {
-            await _schoolService.Create(requestModel.Name, requestModel.Address, requestModel.PrincipalId, requestModel.PrincipalFirstName, requestModel.PrincipalLastName);
+            if (this.User.Identity.IsAuthenticated) return Unauthorized();
+
+            if (!this.ModelState.IsValid) return BadRequest(ModelState);
+
+            var mappedSchoolModel = _mapper
+                    .Map<School>(requestModel);
+
+            await _schoolService.Create(mappedSchoolModel);
 
             return Ok();
         }
 
         /// <summary>
         /// Updates an object of type School and stores it in the database.
         /// </summary>
+        /// <param name="id"></param>
         /// <param name="requestModel"></param>
         /// <returns></returns>
+        [Authorize]
         [HttpPut("Update")]
-        public async Task<IActionResult> Update(SchoolUpdateRequestModel requestModel)
+        public async Task<IActionResult> Update(int id, SchoolUpdateRequestModel requestModel)
```

Comment:
Is there a special reason why the Id is here? It should be inside the Request Model